src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php

```diff
@@ -118,7 +118,7 @@ abstract class FunctionLikeAnalyzer extends SourceAnalyzer
     /**
      * @param Closure|Function_|ClassMethod|ArrowFunction $function
      */
-    protected function __construct($function, SourceAnalyzer $source, FunctionLikeStorage $storage)
+    public function __construct($function, SourceAnalyzer $source, FunctionLikeStorage $storage)
```

Comment:
each child redefine the visibility of the constructor as public. The parent should have the same visibility